docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: "Modifying existing code to work with Dagster Pipes | Dagster Docs"
+description: "This page shows ways to execute external code with Dagster Pipes with different entities in the Dagster systen."
+---
+
+# Reference
+
+This reference shows usage of Dagster Pipes with other entities in the Dagster system. For a step-by-step walkthrough, refer to the [Dagster Pipes tutorial](/guides/dagster-pipes/subprocess).
+
+## Specifying environment variables and extras
+
+When launching the subprocess, you may want to make environment variables or additoional parameters available in the external process. Extras can be arbitrary, user-defined parameters made available on the context object in the external process.
+
+<TabGroup>
+<TabItem name="Provide in the Dagster code">
+
+The `run` method to the `PipesSubprocessClient` resource also accepts `env` and `extras` , which allow you to specify environment variables and extra arguments when executing the subprocess:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/extras_env_dagster_code.py
+import os
+import shutil
+
+from dagster import (
+    AssetExecutionContext,
+    PipesSubprocessClient,
+    asset,
+    file_relative_path,
+)
+
+
+@asset
+def subprocess_asset(
+    context: AssetExecutionContext, pipes_subprocess_client: PipesSubprocessClient
+):
+    cmd = [shutil.which("python"), file_relative_path(__file__, "external_code.py")]
+    return pipes_subprocess_client.run(
+        command=cmd,
+        context=context,
+        extras={"foo": "bar"},
+        env={
+            "MY_ENV_VAR_IN_SUBPROCESS": os.environ["MY_ENV_VAR"],
+        },
+    ).get_materialize_result()
+```
+
+</TabItem>
+<TabItem name="Access in the external code">
+
+In the external code, you can access extras via the `PipesContext` object:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/extras_env_external_code.py lines=2-
+import os
+
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    # get all extras provided by Dagster asset
+    print(context.extras)
+    # get the value of an extra
+    print(context.get_extra("foo"))
+    # get env var
+    print(os.environ["MY_ENV_VAR_IN_SUBPROCESS"])
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+</TabItem>
+</TabGroup>
+
+## Working with @asset_check
```

Comment:
```suggestion
---

## Working with @asset_check
```

Why the decorator in the heading and not `asset checks`?

Reply:
because this is explicitly for working with the `@asset_check` decorator. the usage is different when "asset checks" are defined on `@asset`.

Reply:
Ahh, gotcha